src/System.Security.Cryptography.Primitives/tests/CryptoStream.cs

```diff
@@ -172,20 +172,22 @@ public static void Clear()
             }
         }
 
-        [Fact]
-        public static void FlushAsync()
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public static void FlushAsync(bool cancel)
         {
             ICryptoTransform encryptor = new IdentityTransform(1, 1, true);
             using (MemoryStream output = new MemoryStream())
             using (CryptoStream encryptStream = new CryptoStream(output, encryptor, CryptoStreamMode.Write))
             {
                 encryptStream.WriteAsync(new byte[] { 1, 2, 3, 4, 5 }, 0, 5);
-                Task waitable = encryptStream.FlushAsync(new Threading.CancellationToken(false));
-                Assert.False(waitable.IsCanceled);
-
-                encryptStream.WriteAsync(new byte[] { 1, 2, 3, 4, 5 }, 0, 5);
-                waitable = encryptStream.FlushAsync(new Threading.CancellationToken(true));
-                Assert.True(waitable.IsCanceled);
+                Task waitable = encryptStream.FlushAsync(new Threading.CancellationToken(cancel));
+                Assert.Equal(cancel, waitable.IsCanceled);
+                if (!cancel)
+                {
+                    waitable.Wait();
+                }
```

Comment:
* Presumably `if (cancel)`'s task should have thrown a cancellation exception?
* Why Wait instead of await (xunit supports async test methods)?

So the problem before was that Flush(Cancel=false) was finishing and Flush(Cancel=true) then double-flushed?  Does that mean we're not checking cancellation somewhere we should have? (CryptoStream, the underlying stream, whatever)